compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp

```diff
@@ -163,13 +163,25 @@ enumerateMatmulTileX86_64(EncodingUser user, TypeRange elementTypes,
     // reconsider when taking advantage of native f16/bf16 arithmetic when the
     // accumulator itself is f16/bf16.
     if (hasFeature(target, "+avx512f")) {
-      return {
-          TileMxNxK{16, 16, 1}, // Aim to use VFMADD* (zmm).
-          TileMxNxK{8, 16, 1},  // Truncation of the above.
-          TileMxNxK{4, 16, 1},  // Truncation of the above.
-          TileMxNxK{2, 16, 1},  // Truncation of the above.
-          TileMxNxK{1, 16, 1},  // Truncation of the above.
-      };
+      if (hasUkernel(target)) {
+        return {
+            TileMxNxK{16, 16, 1}, // Aim to use VFMADD* (zmm).
+            TileMxNxK{8, 16, 1},  // Truncation of the above.
+            TileMxNxK{4, 16, 1},  // Truncation of the above.
+            TileMxNxK{2, 16, 1},  // Truncation of the above.
+            TileMxNxK{1, 16, 1},  // Truncation of the above.
+        };
+      } else {
+        // Code generation tile sizes. We try to keep certain level of unrolling
+        // on the N dimension when the M dimension is too narrow for it.
+        return {
+            TileMxNxK{16, 16, 1}, // Aim to use VFMADD* (zmm).
+            TileMxNxK{8, 32, 1},  // Truncation of the above.
+            TileMxNxK{4, 64, 1},  // Truncation of the above.
+            TileMxNxK{2, 64, 1},  // Truncation of the above.
+            TileMxNxK{1, 128, 1}, // Truncation of the above.
```

Comment:
Please only comment `// Truncation of the above` when it really is a truncation of the above. Here, going from `{16, 16, 1}` is not, since it gets wider in `N0`. So you have 2x narrower M0 but 2x wider N0, so still 16 `zmm` accumulator registers.

Reply:
(Which is what the comment just above is saying).